src/spec-node/containerFeatures.ts

```diff
@@ -294,6 +294,14 @@ async function getFeaturesBuildOptions(params: DockerResolverParameters, devCont
 		.replace('#{copyFeatureBuildStages}', getCopyFeatureBuildStages(featuresConfig, buildStageScripts))
 		.replace('#{devcontainerMetadata}', getDevcontainerMetadataLabel(imageMetadata, common.experimentalImageMetadata))
 		;
+	
+	const rawContainerEnv = imageMetadata.raw.find(entry => entry.containerEnv)?.containerEnv;
+	if (rawContainerEnv !== undefined) {
+		let keys = Object.keys(rawContainerEnv);
+		const concatenatedEnv = keys.map(k => `ENV ${k}=${rawContainerEnv![k]}`).join('\n');
+		dockerfile += concatenatedEnv;
```

Comment:
Maybe add a placeholder in `getContainerFeaturesBaseDockerFile()` and add a `replace` line above instead to use the same pattern as above?

Reply:
Updated